Packs/CommonScripts/Scripts/script-ShowOnMap.yml

```
@@ -4,43 +4,60 @@ commonfields:
 name: ShowOnMap
 script: >
   var lat = parseFloat(args.lat);
-
   var lng = parseFloat(args.lng);
+  var address = args.address;
 
+  if (isNaN(lat) || isNaN(lng)) {
+    if (!address || address.trim().length === 0){
+    return {
+        ContentsFormat: formats.text,
+        Type: entryTypes.error,
+        Contents: "Please enter either coordinates (lat,lng) or an address (which requires a GoogleMaps integration instance)."
+    };
 
-  if(isNaN(lat) || isNaN(lng)) {
-      return {
-          ContentsFormat: formats.text,
-          Type: entryTypes.error,
-          Contents: 'Invalid coordinates'
-      };
+    }
+    // An active instance of GoogleMaps exists, geocoding is possible
+      var commandResult = executeCommand("google-maps-geocode", {"search_address" : address});
+      if (isError(commandResult)) {
+          return commandResult;
+      }
+      if (!commandResult.results){
+          return commandResult; // a non-error note indicating no places were found
+      }
```

@dorschw I now understand what you're trying to achieve here. You can unite the conditions.
```suggestion
      if (isError(commandResult) || !commandResult || !commandResult.results) {
          return commandResult;
      }
```